client/network/sync/src/lib.rs

```diff
@@ -631,17 +632,17 @@ where
 					},
 				);
 
-				if let SyncMode::Warp = &self.mode {
+				if let SyncMode::Warp = self.mode {
 					if self.peers.len() >= MIN_PEERS_TO_START_WARP_SYNC && self.warp_sync.is_none()
 					{
 						log::debug!(target: "sync", "Starting warp state sync.");
-						if let Some(provider) = &self.warp_sync_provider {
-							self.warp_sync =
-								Some(WarpSync::new(self.client.clone(), provider.clone()));
-						}
+						self.warp_sync = Some(WarpSync::new(
+							self.client.clone(),
+							self.warp_sync_params.as_mut().unwrap(),
+							None,
+						));
```

Comment:
```suggestion
						if let Some(params) = self.warp_sync_params.take() {
							self.warp_sync = Some(WarpSync::new(
								self.client.clone(),
								params,
								None,
							));
						}
```

Never use `unwrap`! You can use `expect`, but then with a proof why this can not fail and here you could not proof that it would not fail.